music21/duration.py

```diff
@@ -2786,6 +2787,7 @@ def _getQuarterLength(self) -> OffsetQL:
         return self._qtrLength
 
     def _setQuarterLength(self, value: OffsetQLIn):
+        self.expressionIsInferred = True
```

Comment:
if a duration is unlinked that's almost certainly because it has an dedicated expression.  So if I do this:

```
d = duration.Duration()
d.linked = False
d.quarterLength = 1.95
d.type = 'half'
```

I expect that to be the same as:

```
d = duration.Duration()
d.linked = False
d.type = 'half'
d.quarterLength = 1.95
```

Also I don't think that we should change expressionIsInferred if the quarterLength does not change:

```
d = duration.Duration(type='quarter')
d.quarterLength = 1.0
```

shouldn't set expressionIsInferred.

Reply:
Requesting this to be changed back, but THANKS for other parts!

Reply:
good point!

Reply:
then that probably needs to be handled in the cases I added elsewhere, in the qL setter